downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

```diff
@@ -70,6 +96,49 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](
 
 #### OpenAPI tool
 
+#### Persist tool
+- Modified the `persist init` command to solely create a `persist` directory within the Ballerina project and generate a new definition file (`model.bal`) within the `persist` directory if it doesn't already exist. It no longer updates the `Ballerina.toml` file with the persist configuration as it did previously.
+- Modified the `persist generate` command to function as a one-time source code generation tool. Additionally, introduced the following new options to the `persist generate` command:
+    - `--datastore` - This is used to indicate the preferred data store.
+    - `--module` - This is used to indicate the module in which the files are generated.
+
+    For example,
+    ```
+    $ bal persist generate --datastore mysql --module db
+    ```
+
+- Changed the `persist generate` command to generate all the Ballerina types, client, `db_config` files, and the script file in the specified module directory, allowing the user to commit the generated source code along with the project source code.
+- Introduced the new `persist add` command to initialize the `bal persist` feature in the Ballerina project and integrate the source code generation with the `bal build` command. This command will update the `Ballerina.toml` file with the `tool.persist` configuration and generate the `model.bal` file in the `persist` directory, if the file does not exist. This command supports the following options:
+    - `--datastore` - This is used to indicate the preferred data store.
+    - `--module` - This is used to indicate the module in which the files are generated.
+
+    For example,
+    ```
+    $ bal persist add --datastore mysql --module db
+    ```
+
+- Implemented introspection support for existing databases to facilitate the generation of the persist data model. This functionality is accessible through the new `bal persist pull` command. The command is equipped with the following options:
+    - `--datastore` - This is used to indicate the preferred data store.
+    - `--host` - This is used to indicate the host of the database.
+    - `--port` - This is used to indicate the port of the database.
+    - `--user` - This is used to indicate the username of the database.
+    - `--database` - This is used to indicate the database name.
+
+    For example,
+    ```
+    $ bal persist pull --datastore mysql --host localhost --port 3306 --user root --database test
+    ```
+    >**Info:** The database introspection support is an experimental feature and currently only supports MySQL databases. The commands associated with the feature might change in future releases.
+
+- Revised the persist migrate command to extract the datastore configuration from the provided argument instead of the Ballerina.toml file. The command now accepts the following option:
```

Comment:
```suggestion
- Revised the persist migrate command to extract the datastore configuration from the provided argument instead of the `Ballerina.toml` file. The command now accepts the following option:
```